tests/test_proxy_functional.py

```diff
@@ -3,13 +3,47 @@
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def secure_proxy_url(tls_ca_certificate_pem_path, tls_certificate_private_key_path):
+    """Return the an instance of a running secure proxy."""
+    port = str(proxy.common.utils.get_available_port())
+    address = "127.0.0.1"
+    input_args = [
+        "--num-workers",
+        "1",
+        "--threadless",
+        "--hostname",
+        "127.0.0.1",
+        "--port",
+        port,
```

Comment:
Per earlier discussion:
```suggestion
def secure_proxy_url(aiohttp_unused_port, tls_ca_certificate_pem_path, tls_certificate_private_key_path):
    """Return the an instance of a running secure proxy."""
    address = "127.0.0.1"
    input_args = [
        "--num-workers",
        "1",
        "--threadless",
        "--hostname",
        "127.0.0.1",
        "--port",
        str(aiohttp_unused_port),
```